raster-foundry/raster-foundry
# Project analysis edit modals

## Overview
Allow editing analysis input and const nodes from the project/analyses view

Create a generic list-item component and start using it in places

### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

### Demo
![image](https://user-images.githubusercontent.com/4392704/53516920-75ab1900-3a9b-11e9-9d52-46a185863d34.png)

### Notes
We don't set a render definition, so the tiles are still using the default tile-local viridis color scheme and look weird. That is coverd by #4666

## Testing Instructions
* Add a multipolygon geometry to a layer manually which covers an area of the layer which has data
 * Select 2-3 layers including the layer you edited, and create analyses in a batch for them in the project layers view
* Using the same layers, create another set of analyses using a different template
* In the analyses list view, verify that you can only select multiple analyses which were created from the same template
* Edit them as a group and individually, setting the band numbers then previewing them to verify that they change
* Verify that if editing one of the analyses which was created with a mask node, the edit modal shows the geojson on a small map but does not let you edit it (possible future work)
* verify that layer-items that were replaced with list-items have preserved functionality

Closes #4540
